tests/problems/cross_modality/test_translation_problem.py

```diff
@@ -0,0 +1,163 @@
+        tp = TranslationProblem(adata_src, adata_tgt)
+        assert tp.problems == {}
+        assert tp.solutions == {}
+
+        tp = tp.prepare(batch_key="batch", src_attr=src_attr, tgt_attr=tgt_attr, joint_attr=joint_attr)
+
+        assert len(tp) == len(expected_keys)
+        for prob_key in expected_keys:
+            assert isinstance(tp[prob_key], tp._base_problem_type)
+            assert tp[prob_key].shape == (n_obs, n_obs)
+            assert tp[prob_key].x.data_src.shape == (n_obs, x_n_var)
+            assert tp[prob_key].y.data_src.shape == (n_obs, y_n_var)
+            if joint_attr is not None:
+                assert tp[prob_key].xy.data_src.shape == tp[prob_key].xy.data_tgt.shape == (n_obs, xy_n_vars)
+
+    @pytest.mark.parametrize(
+        ("epsilon", "alpha", "rank", "initializer"),
+        [(1e-2, 0.9, -1, None), (2, 0.5, -1, "random"), (2, 0.5, -1, "rank2"), (2, 0.1, -1, None)],
+    )
+    @pytest.mark.parametrize("src_attr", ["emb_src", {"attr": "obsm", "key": "emb_src"}])
+    @pytest.mark.parametrize("tgt_attr", ["emb_tgt", {"attr": "obsm", "key": "emb_tgt"}])
+    def test_solve_balanced(
+        self,
+        adata_translation: AnnData,
+        epsilon: float,
+        alpha: float,
+        rank: int,
+        src_attr: Mapping[str, str],
+        tgt_attr: Mapping[str, str],
+        initializer: Optional[Literal["random", "rank2"]],
+    ):
+        adata_tgt, adata_src = _adata_modality_split(adata_translation)
+        kwargs = {}
+        expected_keys = {(i, "ref") for i in adata_src.obs.batch.cat.categories}
+        if rank > -1:
+            kwargs["initializer"] = initializer
+
+        tp = TranslationProblem(adata_src, adata_tgt)
+        tp = tp.prepare(batch_key="batch", src_attr=src_attr, tgt_attr=tgt_attr)
+        tp = tp.solve(epsilon=epsilon, alpha=alpha, rank=rank, **kwargs)
+
+        for key, subsol in tp.solutions.items():
+            assert isinstance(subsol, BaseSolverOutput)
+            assert key in expected_keys
+            assert tp[key].solution.rank == rank
+
+        assert np.all([np.all(~np.isnan(sol.transport_matrix)) for sol in tp.solutions.values()])
+
+    @pytest.mark.parametrize("args_to_check", [fgw_args_1, fgw_args_2])
+    def test_pass_arguments(self, adata_translation: AnnData, args_to_check: Mapping[str, Any]):
+        adata_tgt, adata_src = _adata_modality_split(adata_translation)
+        tp = TranslationProblem(adata_src, adata_tgt)
+
+        adata_src = adata_src[adata_src.obs["batch"] == 1]
+
+        key = ("1", "ref")
+        tp = tp.prepare(
+            batch_key="batch",
+            src_attr={"attr": "obsm", "key": "emb_src"},
+            tgt_attr={"attr": "obsm", "key": "emb_tgt"},
+            joint_attr="X_pca",
+        )
+        tp = tp.solve(**args_to_check)
+
+        solver = tp[key].solver.solver
+        for arg, val in gw_solver_args.items():
+            assert hasattr(solver, val)
```

Comment:
No need for this, the below `getattr` without a default will fail (also in other places).